com.github.fries1234.ncsa-mosaic.json

```diff
@@ -0,0 +1,112 @@
+            ]
+        },
+        {
+            "name": "libxmu",
+            "buildsystem": "autotools",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://xorg.freedesktop.org/releases/individual/lib/libXmu-1.1.3.tar.bz2",
+                    "sha256": "9c343225e7c3dc0904f2122b562278da5fed639b1b5e880d25111561bac5b731"
+                }
+            ]
+        },
+        {
+            "name": "libxft",
+            "buildsystem": "autotools",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://xorg.freedesktop.org/releases/individual/lib/libXft-2.3.4.tar.bz2",
+                    "sha256": "57dedaab20914002146bdae0cb0c769ba3f75214c4c91bd2613d6ef79fc9abdd"
+                }
+            ]
+        },
+        {
+            "name": "openmotif",
+            "buildsystem": "autotools",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://downloads.sourceforge.net/project/motif/Motif%202.3.8%20Source%20Code/motif-2.3.8.tar.gz",
+                    "sha256": "859b723666eeac7df018209d66045c9853b50b4218cecadb794e2359619ebce7"
+
+                }
+            ],
+            "post-install": [
+                "rm -rf /app/lib/*.a"
+            ]
+        },
+        {
+            "name": "mosaic",
+            "buildsystem": "simple",
+            "build-commands": [
+                "CFLAGS=-fcommon meson build",
+                "ninja -C build",
+                "install -Dm755 build/src/Mosaic -t /app/bin",
+                "install -Dm644 com.github.fries1234.ncsa-mosaic.desktop /app/share/applications/com.github.fries1234.ncsa-mosaic.desktop",
+                "install -Dm644 com.github.fries1234.ncsa-mosaic.appdata.xml /app/share/metainfo/com.github.fries1234.ncsa-mosaic.appdata.xml",
+                "install -Dm644 com.github.fries1234.ncsa-mosaic-symbolic.svg /app/share/icons/hicolor/symbolic/apps/com.github.fries1234.ncsa-mosaic-symbolic.svg",
+                "install -Dm644 com.github.fries1234.ncsa-mosaic.png /app/share/icons/hicolor/256x256/apps/com.github.fries1234.ncsa-mosaic.png",
+                "install -Dm644 com.github.fries1234.ncsa-mosaic.128px.png /app/share/icons/hicolor/128x128/apps/com.github.fries1234.ncsa-mosaic.png",
+                "install -Dm644 com.github.fries1234.ncsa-mosaic.64px.png /app/share/icons/hicolor/64x64/apps/com.github.fries1234.ncsa-mosaic.png"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/fries1234/ncsa-mosaic",
+                    "branch": "master",
+                    "commit": "3fcf1f51568defd0e89b3c134f129634ad7dbbed"
+                },
+                {
+                    "type": "file",
+                    "path": "com.github.fries1234.ncsa-mosaic.desktop"
```

Comment:
desktop, appdata and icons shoud be managed upstream (since you are upstream). None of this is specific flatpak.